test/e2e/framework/framework.go

```diff
@@ -120,11 +120,9 @@ func (f *Framework) WaitForGameServerState(gs *stable.GameServer, state stable.G
 
 		return false, nil
 	})
-	if err != nil {
-		return nil, errors.Wrapf(pollErr, "waiting for GameServer to be %v %v/%v: %v",
-			state, gs.Namespace, gs.Name, err)
-	}
-	return readyGs, nil
+
+	return readyGs, errors.Wrapf(err, "waiting for GameServer to be %v %v/%v",
+		state, gs.Namespace, gs.Name)
```

Comment:
I don't think this would replace your previous change. pollErr is replaced with err which is local to the PollImmediate method. 
err := wait.PollImmediate --> this error will still be timeout as it is different that the err inside PollImmediate.

Reply:
Yeah, that was my intent with this change. I'm saying I don't think we need to do replace err with pollErr at all.

The original code was in place to make the pollErr error visible, in case it is the cause of the test failing -- but we already have logging in place to show that error, so it seems redundant to do do both logging _and_ bubble the error up.

If we remove the bubble up, the code is cleaner, and we don't sacrifice the ability to discover the reasons for issues.

At least - that is my intent :smile: